src/SettingsSystem.cpp

```diff
@@ -0,0 +1,121 @@
+#include "SettingsSystem.h"
+#include "InputSettings.h"
+#include "AudioSettings.h"
+#include <SDL.h>
+
+namespace {
+    using Key = const std::string; // Convenience typedef
```

Comment:
I'd argue that typedefs increase clarity because it's a little more clear how the value will be passed around and used. Usually I'd use the using keywords for things that are more obnoxious but I see no problem with this specific usage.

Reply:
Just use the old syntax for typedef then. They are equivalent yet avoids "using". Best of both worlds

Reply:
Excellent idea. I 100% approve of such a change.